tests/e2e-tests/specs/admin/shipping-settings.test.js

```diff
@@ -0,0 +1,348 @@
+ */
+import { StoreOwnerFlow } from "../../utils/flows";
+import { AccountWithNoCreditCard, AccountWithOneCreditCard, AccountWithTwoCreditCard, AccountWithTwoCreditCardAndNoDefault } from "../../fixtures/account_settings";
+
+// Click save and wait until it's saved.
+const saveAndWait = async () => {
+    await expect( page ).toClick( '.button.is-primary', { text: 'Save changes' } );
+    await page.waitForSelector('[class="button is-primary"]');
+};
+
+/**
+ * This function will wait for a button with any CSS selector + text value.
+ *
+ * @param {string} selector CSS selector
+ * @param {string} text The text value of the element we want to search for. ie. button's value, div's innertext.
+ */
+const waitForSelectorAndText = async (selector, text) => {
+    return await page.waitForFunction(
+        (cssSelector, innerTextContent) => !!Array.from(document.querySelectorAll(cssSelector)).find(el => el.textContent.trim() === innerTextContent.trim()),
+        {},
+        selector, text
+    );
+};
+
+describe( 'Saving shipping label settings', () => {
+	it( 'Can toggle shipping labels' , async () => {
+		await StoreOwnerFlow.login();
+		await StoreOwnerFlow.openSettings('shipping', 'woocommerce-services-settings');
+        await page.waitForSelector('.form-toggle__switch');
+        await expect(page).toClick('.form-toggle__switch');
+        await page.waitForSelector('.card.label-settings__labels-container', {
+            visible: false
+        });
+        await expect( page ).toClick( '.button.is-primary', { text: 'Save changes' } );
+        // Clicking save should persist the visible state of the section.
+        await page.waitForSelector('.card.label-settings__labels-container', {
+            visible: false
+        });
+
+        // Toggle it back to enable so the following tests can run.
+        await expect(page).toClick('.form-toggle__switch');
+        await saveAndWait();
+    });
+
+    it ('Should be able to select a different paper size', async () => {
+        // Save it as legal
+        await page.select('select.form-select', 'legal');
+        await saveAndWait();
+        await expect(page).toMatchElement('.notice.is-success .notice__text', { text: 'Your shipping settings have been saved.' });
+
+        let paperSize = await page.$('select.form-select');
+        let selectedOption = await (await paperSize.getProperty('value')).jsonValue();
+        expect(selectedOption).toBe('legal');
+
+        // Save it back to label
+        await page.select('select.form-select', 'label');
+        await saveAndWait();
+        await expect(page).toMatchElement('.notice.is-success .notice__text', { text: 'Your shipping settings have been saved.' });
+
+        paperSize = await page.$('select.form-select');
+        selectedOption = await (await paperSize.getProperty('value')).jsonValue();
+        expect(selectedOption).toBe('label');
+    });
+} );
+
+describe( 'Shipping label payment method', () => {
```

Comment:
For all test cases in this `describe`, intercept all API to `wp-json/wc/v1/connect/account/settings` and return a mocked response. 
Remove this interception after this test case.